src-django/api/models.py

```diff
@@ -97,3 +97,70 @@ def save(self, **kwargs):
 
     class Meta:
         app_label = 'api'
+
+
+class Condition(models.Model):
+    CRITERIA_TYPES = (
+        ('EQUALS', 'EQUALS'),
+        ('GREATER', 'GREATER'),
+        ('LESSER', 'LESSER')
+    )
+
+    NODE_TYPES = (
+        ('AND', 'AND'),
+        ('OR', 'OR'),
+        ('NOT', 'NOT'),
+        ('CRITERIA', 'CRITERIA')
+    )
+
+    parent = models.ForeignKey('self', on_delete=models.CASCADE, related_name='children', blank=True, null=True)
+    element = models.ForeignKey(Element, blank=True, null=True)
+    node_type = models.CharField(max_length=8, choices=NODE_TYPES, null=True, blank=True)
+    criteria_type = models.CharField(max_length=8, choices=CRITERIA_TYPES, null=True, blank=True)
+    value = models.TextField(blank=True, null=True)
+    last_modified = models.DateTimeField(auto_now=True)
+    created = models.DateTimeField(auto_now_add=True)
+
+    def save(self, **kwargs):
+        if self.node_type:
```

Comment:
remember that we are exposing the API to the public so it's always good to have as much sanity checks as possible